cli/cmd/upgrade/upgrade.go

```diff
@@ -374,9 +331,6 @@ func (c *Command) validateFlags(args []string) error {
 		}
 	}
 
-	if c.flagDryRun {
-		c.UI.Output("Performing dry run upgrade.", terminal.WithInfoStyle())
```

Comment:
Do we want to print something like this at the top before anything else so the user knows all their output is from a dry run?

Reply:
Sure, I just deleted it here because it was printing twice. Once here and once later on. I can shift that single printing to be earlier.

Reply:
I think it would ease user anxiety to show this earlier so I will.

Reply:
I added it on line 178 to print "Dry Run Upgrade: No changes will be made to the cluster."

I didn't want it to be in `validateFlags` because it's a side effect.